### PR TITLE
Widget

### LINKED ISSUE #16: Components/widget

### PR DESCRIPTION
**Pull request by halimahexe**
## Description
Created the widget card and header card for use on dashboard.
There are two options for the WidgetHeader, with orange badge and without:
image

## Related Issue
Closes #16 

## How Can This Be Tested?
Run `npm install` and `npm run dev` and view on the homepage. To test the content of the widget, you can pass in one of the following categories: ```'savings', 'expenses', 'investing', 'charity'```, change the text passed into `description` or set `isCoreTask={true}`.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings, including `` npm run build ``
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

File: src/components/Widget/Widget.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import {
+	FaCloudRain,
+	FaMoneyBill,
+	FaHourglassHalf,
+	FaSeedling,
+	FaStar,
+} from 'react-icons/fa6';
+import { Badge } from '@/components/ui/badge';
+import { IconType } from 'react-icons/lib';
+import { string } from 'zod';
```
Comment (oskarprzybylski23) [inline src/components/Widget/Widget.tsx:10]: It seems like this is not used? Can you please remove if not needed?

Comment (halimahexe) [reply src/components/Widget/Widget.tsx]: Thanks for flagging, hadn't spotted that!

**Comment by halimahexe** [comment]
I've added some functionality to the widgets, to show when goals are active/inactive. As you can see below, the greyed out widget is done, there's a hover effect which makes the background purple and the default state is an off-white background:

image

**Review by oskarprzybylski23 — CHANGES_REQUESTED** [review]
Thanks. Widgets look great!
just two things to address:

I had an issue that it wouldn't build because of the error below:
image

Also, could you remove widgets from Home page now that they are finished, just not to conflict with what we already have there.

**Comment by halimahexe** [comment]
Ah thanks for flagging. Totally forgot to run `npm run build` on my latest changes.

**Comment by halimahexe** [comment]
Thanks for flagging. Both fixed!

Here's the code for displaying widgets on their proper page, in case helpful:
```ts
<div className='grid grid-cols-2 lg:grid-cols-4 gap-1 items-start justify-items-center'>
				<WidgetHeader category='savings' heading='Emergency Savings'>
					<Widget
						category='savings'
						description="I have at least 50% of one month's salary saved"
						isCoreTask={true}
					/>
				</WidgetHeader>
				<WidgetHeader category='expenses' heading='Managing Expenses'>
					<Widget
						category='expenses'
						description='I can login online to all my utility providers'
						isCoreTask={false}
					/>
				</WidgetHeader>
				<WidgetHeader category='investing' heading='Investing in the Future'>
					<Widget
						category='investing'
						description='I have a list of any outstanding debts'
						isCoreTask={false}
					/>
				</WidgetHeader>
				<WidgetHeader category='charity' heading='Giving Back'>
					<Widget
						category='charity'
						description='I purify any interest I receive in my bank accounts'
						isCoreTask={true}
					/>
					<Widget
						category='charity'
						description='I give back in person'
						isCoreTask={false}
					/>
				</WidgetHeader>
			</div>
```

**Review by oskarprzybylski23 — APPROVED** [review]
Thanks for addressing, LGTM!

**Pull request merged**